slapo/schedule.py

```diff
@@ -790,25 +791,33 @@ def is_module_list(module):
     return root_sch
 
 
-def consolidate_model(sch: Schedule, topology=None):
+def consolidate_model(
+    sch: Schedule,
+    topology=None,
+    param_init_fn: Optional[Callable[[nn.Module], None]] = None,
+):
+    if dist.get_world_size() > sch.world_size:
```

Comment:
The use of `dist` in this function makes `consolifate_model` require `dist` to be initialized. In other words, single device use cases will crash right after calling this function. Since now whether to call this function is by user config, we have to provide a single device fallback.

Reply:
makes sense